### ISSUE TITLE
Do not download non-preview-able data when generating link previews

### ISSUE THREAD

**Issue by nathanlesage** [Zettlr/Zettlr#4765]
### Discussed in https://github.com/Zettlr/Zettlr/discussions/4762

I am opening this issue here as discussed in the linked discussion. The basic problem is the following:

* In v3.0.0 I implemented link previews (akin to Twitter's link cards) to give users a preview of what a link is without them having to click it.
* This is great for any type of website, and even images (since that's only text or a few kb of image data)
* However, the way I implemented this, it will *always force-download even files before checking if Zettlr can even generate a preview*

In other words, we'll have to split up the link preview fetcher: First, only "knock on the door" of the webserver and inspect the header, determining what type of content comes (inspect the mime-type). If the mime-type is supported, generate a preview, otherwise don't. This should prevent (as in the issue) downloading prohibitively large PDF files.

**Comment by nathanlesage** [comment]
Helpful comment that shows how to cancel a request using `got` after receiving the headers (and inspecting them):

https://github.com/sindresorhus/got/issues/771#issuecomment-481788301

**Comment by nathanlesage** [comment]
@Phillh56 I have now added some functionality that will first "probe" the link and look at the headers. If the content cannot be previewed (e.g., for a PDF file, because that would require downloading the entire thing), it will immediately stop and not try to download the entire thing anymore. In other words, per link preview, Zettlr will maybe download a few kilobytes instead of several Gigabytes worth of PDF.

I will soon-ish release v3.0.3 that will include this fix. Please then re-activate the setting and monitor Zettlr's data usage. If it's still very low even with preview generation on, that's good! If it starts to creep up again, please comment here so that I can take another look!
